File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(computers,i) :
    if '*' in computers[i] :
        for j in range(0,len(computers[i])) :
            if computers[i][j] == 1 :
                computers[i][j] ='*'
                computers[j][i] ='*'
                dfs(computers,j)

def solution(n, computers):
    answer = 0
    for i in range(0,n) :
        if 1 in computers[i] and '*' not in computers[i]  :
            answer+=1
            for k in range(0,n) :              
                if computers[i][k] ==1 :
                    computers[i][k] = '*'
                    computers[k][i] = '*'
                    dfs(computers,k)
            
    print(computers)
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] com = new boolean[n];
        
        
        
        for(int i =0; i<computers.length; i++){
            if(!com[i]){
                answer++;
                dfs(i, computers, com);
            } 
        }
        
        return answer;
    }
    
    public void dfs(int node, int[][] computers, boolean[] com){
        com[node] = true;
        
        for(int j=0; j<computers[node].length; j++){
            if(computers[node][j] == 1 && !com[j]){
                com[j] = true;
                dfs(j, computers, com);
            }
        }
    }
}